src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs

```diff
@@ -578,10 +578,52 @@ public static IEnumerable<string> SplitCommandLineIntoArguments(string commandLi
                 }))
             .Select(arg => arg.Trim())                                                                  // Trim whitespace
             .TakeWhile(arg => (!removeHashComments || !arg.StartsWith("#", StringComparison.Ordinal)))  // If removeHashComments is true, skip all arguments after one that starts with '#'
-            .Select(arg => CondenseDoubledBackslashes(arg).Unquote())                                   // Remove quotes and handle backslashes.
+            .Select(arg => UnquoteAndUnescape(NormalizeBackslashes(arg)))                                 // Remove quotes and handle backslashes.
             .Where(arg => !string.IsNullOrEmpty(arg));                        							// Don't produce empty strings.
         }
 
+        private static string UnquoteAndUnescape(string v)
```

Comment:
A lot of the behavior here seems similar to the split in SplitCommandLineIntoArguments. Can we be clearer here about what the preconditions/postconditions are for this method?

How do we expect the command line to change after the first split and what do we expect it to look like after UnquoteAndUnescape?